wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -691,6 +699,90 @@ protected function loadLanguageIDs() {
 		}
 	}
 	
+	/**
+	 * If multi-factor authentication is enabled for the given user then
+	 * - the userID will be stored in the session variables, and
+	 * - `true` is returned.
+	 * Otherwise,
+	 * - `changeUser()` will be called, and
+	 * - `false` is returned.
+	 * 
+	 * If `true` is returned you should perform a redirect to `MultifactorAuthenticationForm`.
+	 */
+	public function changeUserAfterMultifactor(User $user): bool {
```

Comment:
The method name sounds somehow "wrong". Shouldn't it be called before the multi-factor is active? Unfortunately, I can't think of a better name ad hoc either.

Reply:
The method name is “truncated”. The full name would be `changeUserAfterMultifactorAuthentication` which is unwieldy.

Reply:
If you are concerned about the length, then use `changeUserAfter2FA()` instead, otherwise use the full name. Keep in mind that you read code far more often than write it and 6 months from now nobody knows about this shortcut.

Reply:
If anything it would be `Mfa`, but that is even worse. I appended the `Authentication`.